torch_geometric/data/hetero_data.py

```diff
@@ -580,7 +580,7 @@ def subgraph(self, subset_dict: Dict[NodeType, Tensor]) -> 'HeteroData':
             subset_dict (Dict[str, LongTensor or BoolTensor]): A dictonary
                 holding the nodes to keep for each node type.
         """
-        data = self.__class__(self._global_store)
+        data = self.__class__(**self._global_store)
```

Comment:
Why do we need this change?

Reply:
Since self.__class__ is HeteroExplanation, the dictionary self._global_store is interpreted as x_dict, and it complains for the other arguments (not to mention all the arguments are stored incorrectly in x_dict).

Since HeteroData() treats _mapping and kwargs together, this operation doesn't change the existing behavior and allows the arguments to be correctly mapped for HeteroExplanation.